tests/table/table_style_plugin.test.ts

```diff
@@ -0,0 +1,241 @@
+      styles[row - zone.top].push(cellStyle);
+    }
+  }
+  return styles;
+}
+
+describe("Table style", () => {
+  beforeEach(() => {
+    model = new Model();
+    sheetId = model.getters.getActiveSheetId();
+  });
+
+  describe("Table cell style", () => {
+    const style = TABLE_PRESETS["TableStyleMedium9"];
+    const headerColor = style.headerRow!.style!.fillColor;
+    const tableBackgroundColor = style.wholeTable!.style!.fillColor;
+    const totalColor = style.totalRow!.style!.fillColor;
+    const bandedRowColor = style.firstColumnStripe!.style!.fillColor;
+
+    test("Table style is correctly applied", () => {
+      createTable(model, "A1:A4");
+      updateTableConfig(model, "A1:A4", {
+        styleId: "TableStyleMedium9",
+        numberOfHeaders: 1,
+        totalRow: true,
+        bandedRows: true,
+      });
+      expect(getCellStyle("A1")).toMatchObject({ fillColor: headerColor, bold: true });
+      expect(getCellStyle("A2")).toMatchObject({ fillColor: bandedRowColor });
+      expect(getCellStyle("A3")).toMatchObject({ fillColor: tableBackgroundColor });
+      expect(getCellStyle("A4")).toMatchObject({ fillColor: totalColor });
+    });
+
+    test("Table style do not overwrite cell style", () => {
+      createTable(model, "A1:A4");
+      updateTableConfig(model, "A1:A4", { styleId: "TableStyleMedium9", numberOfHeaders: 1 });
+      setStyle(model, "A1", { fillColor: "#f00" });
+      expect(getCellStyle("A1")).toMatchObject({ fillColor: "#f00" });
+    });
+
+    test("Falsy cell style do not overwrite table", () => {
+      createTable(model, "A1:A4");
+      updateTableConfig(model, "A1:A4", { styleId: "TableStyleMedium9", numberOfHeaders: 1 });
+      setStyle(model, "A1", { fillColor: undefined, bold: false });
+      expect(getCellStyle("A1")).toMatchObject({ fillColor: headerColor, bold: true });
+    });
+
+    test("Table style is applied correctly with hidden headers", () => {
+      createTable(model, "A1:B5");
+      updateTableConfig(model, "A1:B5", {
+        styleId: "TableStyleMedium9",
+        numberOfHeaders: 2,
+        bandedRows: true,
+        firstColumn: true,
+      });
+      hideRows(model, [0, 3]);
+      hideColumns(model, ["A"]);
+
+      expect(getCellStyle("A1")).toEqual({}); // hidden
+      expect(getCellStyle("B1")).toMatchObject({}); // hidden
+      expect(getCellStyle("B2")).toMatchObject({ fillColor: headerColor, bold: true }); // Only one header row, the other is hidden
+      expect(getCellStyle("B3")).toMatchObject({ fillColor: bandedRowColor });
+      expect(getCellStyle("B4")).toEqual({}); // hidden
+      expect(getCellStyle("B5")).toMatchObject({ fillColor: tableBackgroundColor }); // banded color is alternating even if a row is hidden
+    });
+  });
+
+  describe("Table borders", () => {
+    const outerBordersTableStyle = {
+      styleId: "TableStyleLight8",
+      numberOfHeaders: 0,
+      bandedRows: false,
+    };
```

Comment:
To ensure no test mutates this object which would affect other tests in a non-deterministic way
```suggestion
    const outerBordersTableStyle = {
      styleId: "TableStyleLight8",
      numberOfHeaders: 0,
      bandedRows: false,
    } as const;
```